Frontend/ui-customer/src/Helpers/Ride.purs

```diff
@@ -346,223 +348,230 @@ checkRideStatus rideAssigned prioritizeRating = do
+                                  , hasSafetyIssue = hasSafetyIssue'
+                                  , hasTollIssue = hasTollIssue'
+                                  , showIssueBanners = hasAccessibilityIssue' || hasSafetyIssue' || hasTollIssue'
+                                  }
                                 }
-                            , ratingViewState
-                                { rideBookingRes = (RideBookingRes resp)
-                                }
-                            , isSafetyCenterDisabled = state.props.isSafetyCenterDisabled 
-                            , bookingId = resp.id
-                            , additionalCharges = [
-                                {
-                                  text :  getString if ride.tollConfidence == (Just CTA.Unsure) then  TOLL_ROAD_CHANGED else if finalFareHasToll then  TOLL_CHARGES_INCLUDED else TOLL_ROAD_CHANGED 
-                                , visibility : boolToVisibility $ finalFareHasToll || estimateFareHasToll
-                                , image :  fetchImage FF_COMMON_ASSET "ny_ic_grey_toll"
-                                , textColor : Color.black700
-                                },
-                                {
-                                  text : maybe "" (\parking ->  getString $ PARKING_CHARGES_INCLUDED $ (getCurrency appConfig) <>  (show $ INT.ceil $ parking ^. _amount)) parkingCharges
-                                , visibility : boolToVisibility $ isJust parkingCharges 
-                                , image : fetchImage FF_COMMON_ASSET "ny_ic_parking_logo_grey"
-                                , textColor : Color.black700
-                                }
-                              ]
-                            , customerIssue = riderRideCompletedScreen.customerIssue
-                                { showIssueBanners = hasAccessibilityIssue' || hasSafetyIssue' || hasTollIssue'
-                                , hasAccessibilityIssue = hasAccessibilityIssue'
-                                , hasSafetyIssue = hasSafetyIssue'
-                                , hasTollIssue = hasTollIssue'
-                                }
+                              , toll {
+                                confidence = ride.tollConfidence
+                              , showAmbiguousPopUp = ride.tollConfidence == Just CTA.Unsure
+                              }
+                              , requestorPartyRoles = contents.requestorPartyRoles
+                            }
                           }
-                      )
-                updateLocalStage RideCompleted
-              when (length listResp.list == 0 && not state.props.isSharedLocationFlow) $ do 
-                modifyScreenState $ HomeScreenStateType (\homeScreen → homeScreen{props{currentStage = HomeScreen}})
-                updateLocalStage HomeScreen
+                        )
+
+                    modifyScreenState
```

Comment:
Can you move this to some different function, this one function is becoming large